ReactSkia/modules/platform/libcurl/RSkNetworkingModule.cpp

```diff
@@ -255,71 +113,53 @@ jsi::Value RSkNetworkingModule::abortRequest(folly::dynamic requestId) {
     RNS_LOG_ERROR ("networkRequest is not valid \n");
     return jsi::Value((int)CURL_RETURN_FAILURE);
   }
-  curl_easy_cleanup(networkRequest->curl_);
-  networkRequest->curl_  = NULL;
-  connectionList_.erase(requestId.asInt());
+  if(sharedCurlNetworking_->abortRequest(networkRequest->curlRequest)) {
+    RNS_LOG_DEBUG (" aborting Curl is success \n");
+    connectionList_.erase(networkRequest->requestId_);
+  }
   return jsi::Value((int)CURL_RETURN_SUCESS);
 }
 
-void RSkNetworkingModule::sendProgressEventwrapper(NetworkRequest *networkRequest, double dltotal,double dlnow, double ultotal, double ulnow) {
+void RSkNetworkingModule::sendProgressEventwrapper(double dltotal,double dlnow, double ultotal, double ulnow, void *userdata) {
+  struct NetworkRequest *networkRequest = ((struct NetworkRequest *)userdata);
```

Comment:
the typecase is already done in progresscallback , so can we avoid this again here and receive the argument as NetworkRequest type instead of void